From: Kaixu Xia <kaixuxia@tencent.com>
Subject: mm/damon/vaddr: indicate the target is invalid when 'nr_regions' is zero
Date: Tue, 13 Sep 2022 17:11:27 +0800

When 'init()' and 'update()' DAMON operations failed and the number of the
damon_target regions is zero, the kdamond would do nothing to this
monitoring target in this case.  It makes no sense to run kdamond when all
of monitoring targets have no regions.  So add the judgement in
'target_valid()' operation to indicate the target is invalid when
'nr_regions' is zero.

Link: https://lkml.kernel.org/r/1663060287-30201-5-git-send-email-kaixuxia@tencent.com
Signed-off-by: Kaixu Xia <kaixuxia@tencent.com>
Cc: SeongJae Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/damon/vaddr.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/mm/damon/vaddr.c~mm-damon-vaddr-indicate-the-target-is-invalid-when-nr_regions-is-zero
+++ a/mm/damon/vaddr.c
@@ -598,6 +598,9 @@ static bool damon_va_target_valid(void *
 	struct damon_target *t = target;
 	struct task_struct *task;
 
+	if (!damon_nr_regions(t))
+		return false;
+
 	task = damon_get_task_struct(t);
 	if (task) {
 		put_task_struct(task);
_
